xmake/rules/c++/modules/modules_support/dependency_scanner.lua

```diff
@@ -174,22 +174,41 @@ end
 
 -- generate edges for DAG
 function _get_edges(nodes, modules)
-  local edges = {}
-  for _, node in ipairs(nodes) do
-      local module = modules[node]
-      if module.requires then
-          for required_name, _ in table.orderpairs(module.requires) do
-              for _, required_node in ipairs(nodes) do
-                  local name, _, _ = compiler_support.get_provided_module(modules[required_node])
-                  if name and name == required_name then
-                      table.insert(edges, {required_node, node})
-                      break
-                  end
-              end
-          end
-      end
-  end
-  return edges
+    local edges = {}
+    local require_std_compat_module = false
+    for _, node in ipairs(nodes) do
+        local module = modules[node]
+        local module_name, _, _ = compiler_support.get_provided_module(module)
+        if module.requires then
+            for required_name, _ in table.orderpairs(module.requires) do
+                if required_name == "std.compat" then
+                    require_std_compat_module = true
+                    break
+                end
+            end
+        end
+    end
+    for _, node in ipairs(nodes) do
+        local module = modules[node]
+        local module_name, _, cppfile = compiler_support.get_provided_module(module)
+        if module.requires then
+            for required_name, _ in table.orderpairs(module.requires) do
+                for _, required_node in ipairs(nodes) do
```

Comment:
Is there any way to optimize this many loops? it seems like it would be very slow when the object file is very large.

Or we can build the map to optimize it the first time we traverse the nodes.